compiler/desugared/ast.ml

```diff
@@ -174,7 +174,10 @@ let empty_rule
     (parameters : (Uid.MarkedString.info * typ) list Mark.pos option) : rule =
   {
     rule_just = Expr.box (ELit (LBool false), Untyped { pos });
-    rule_cons = Expr.box (EEmptyError, Untyped { pos });
+    rule_cons =
+      Expr.eerroronempty
```

Comment:
I thought you could avoid `ErrorOnEmpty EmptyError` no?

Reply:
in the normal case yes, because we expect a default term to be of type `< 'a >`; 
but when building the exception tree on the lhs of a default term, we may need rules that return a non-default term.

I'll definitely clean this up (removing these empty exceptions that are just leftovers from the construction of the tree seems like the best way) to avoid this `error_empty` appearing at a disturbing place.